src/test/java/org/folio/rest/support/matchers/DomainEventAssertions.java

```diff
@@ -39,7 +39,7 @@ private static void assertCreateEvent(KafkaMessage<JsonObject> createEvent, Json
     assertThat(createEvent.getPayload().getString("type"), is("CREATE"));
     assertThat(createEvent.getPayload().getString("tenant"), is(TENANT_ID));
     assertThat(createEvent.getPayload().getJsonObject("old"), nullValue());
-    assertThat(createEvent.getPayload().getJsonObject("new"), is(newRecord));
+    assertThat(createEvent.getPayload().getJsonObject("new").putNull("_version"), is(newRecord.putNull("_version")));
```

Comment:
assertThat(createEvent.getPayload().getJsonObject("new").remove("_version"), is(newRecord.remove("_version")));

Reply:
Why is there a need to change the JSON being compared during the assertions?

Reply:
> ```
> assertThat(createEvent.getPayload().getJsonObject("new").remove("_version"), is(newRecord.remove("_version")));
> ```

That won't work because "remove" returns the object that is removed or null.

Reply:
> Why is there a need to change the JSON being compared during the assertions?

Because mod-inventory-storage does not read after save.

Reply:
> Because mod-inventory-storage does not read after save.

Is that because we've chosen to use triggers to do the version incrementing and so the code doesn't know about it unless it reads the record from the database after the write?

Reply:
Correct

Reply:
Have we checked with the folks using these messages if version is important (and needs to be provided in them)?

For example, if the instance representation provided by mod-search (one of the clients) will eventually be used for updating instances in ui-inventory then if it does not have the version, the front end won't have it to provide to the server during a change.

Reply:
> Have we checked with the folks using these messages if version is important to them (and hence needs to be provided in them)?
> 
> For example, if the instance representation provided by mod-search (one of the clients) will eventually be used for updating instances in ui-inventory then if it does not have the version, the front end won't have it to provide to the server during a change.

Good point. Hopefully relevant backend dev will comment.

Reply:
@bohdan-suprun @MikhailFokanov Do you have thoughts on this?

Reply:
I have created https://issues.folio.org/browse/MODINVSTOR-662

Reply:
@bohdan-suprun Which means you and @MikhailFokanov consider this behaviour to be acceptable for the moment?

Reply:
Yes, correct.